src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -254,17 +254,11 @@ public boolean getLocationPermissions() {
 		boolean hasCoarseLocation = ContextCompat.checkSelfPermission(this, ACCESS_COARSE_LOCATION) == PERMISSION_GRANTED;
 
 		if (hasFineLocation && hasCoarseLocation) {
-			trace(this, "getLocationPermissions() :: already granted");
+			trace(this, "getLocationPermissions() :: Fine and Coarse location already granted");
 			return true;
 		}
 
-		if (settings.hasUserDeniedGeolocation()) {
-			trace(this, "getLocationPermissions() :: user has previously denied to share location");
-			locationRequestResolved();
-			return false;
-		}
-
-		trace(this, "getLocationPermissions() :: location not granted before, requesting access...");
+		trace(this, "getLocationPermissions() :: Fine or Coarse location not granted before, requesting access...");
```

Comment:
If we're granted with just approximate then let the user continue using the form, next time (this code) try again to get precise location.

<img width="350" src="https://user-images.githubusercontent.com/66472237/163980992-b5a7fbf0-d702-464b-9c69-7dfcf1bc2df6.jpg"/>

Open to suggestions